public/pages/AnomalyCharts/containers/AnomaliesChart.tsx

```diff
@@ -256,183 +177,129 @@ export const AnomaliesChart = React.memo((props: AnomaliesChartProps) => {
-                  isLoading={props.isLoading}
-                />
+          {props.isHCDetector &&
+          props.onHeatmapCellSelected &&
+          props.detectorCategoryField ? (
+            <EuiFlexGroup style={{ padding: '20px' }}>
+              <EuiFlexItem style={{ margin: '0px' }}>
+                <div
+                  style={{
+                    width: '100%',
+                    opacity: showLoader ? 0.2 : 1,
+                  }}
+                >
+                  {props.isLoading ? (
+                    <EuiFlexGroup
+                      justifyContent="spaceAround"
+                      style={{ paddingTop: '150px' }}
+                    >
+                      <EuiFlexItem grow={false}>
+                        <EuiLoadingChart size="xl" mono />
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  ) : (
+                    [
+                      <AnomalyHeatmapChart
+                        detectorId={props.detectorId}
+                        detectorName={props.detectorName}
+                        dateRange={props.dateRange}
+                        //@ts-ignore
+                        title={props.detectorCategoryField[0]}
+                        anomalies={anomalies}
+                        isLoading={props.isLoading}
+                        showAlerts={props.showAlerts}
+                        monitor={props.monitor}
+                        detectorInterval={props.detectorInterval}
+                        unit={props.unit}
+                        onHeatmapCellSelected={props.onHeatmapCellSelected}
+                      />,
+                      props.showAlerts === undefined || !props.showAlerts
+                        ? [
+                            <EuiSpacer size="m" />,
+                            <AnomalyOccurrenceChart
+                              title={
+                                props.selectedHeatmapCell
+                                  ? props.selectedHeatmapCell.entityValue
+                                  : '-'
+                              }
+                              dateRange={props.dateRange}
+                              onDateRangeChange={props.onDateRangeChange}
+                              onZoomRangeChange={props.onZoomRangeChange}
+                              anomalies={anomalies}
+                              bucketizedAnomalies={false}
+                              anomalySummary={INITIAL_ANOMALY_SUMMARY}
+                              isLoading={props.isLoading}
+                              anomalyGradeSeriesName="Anomaly grade"
+                              confidenceSeriesName="Confidence"
+                              showAlerts={false}
+                              detectorId={props.detectorId}
+                              detectorName={props.detectorName}
+                              detector={props.detector}
+                              detectorInterval={get(
```

Comment:
We have passed `detector`, why still pass `detectorInterval` ? We can get `detectorInterval` from detector.

Reply:
same with id and name

Reply:
Good catch. will change it.

Reply:
refactored